ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -329,6 +335,8 @@ bracketChainSyncClient ::
  -> StrictTVar m (Map peer (ChainSyncClientHandle m blk))
     -- ^ The kill handle and states for each peer, we need the whole map because we
     -- (de)register nodes (@peer@).
+ -> STM m GsmState
+    -- ^ A function giving the current GSM state; only used at startup.
```

Comment:
If it is only used once, shouldn't we take a `GsmState` instead of a `STM m GsmState`?

Reply:
We could, but if we do I'm afraid that the `GsmState` may change between the time we call `bracketChainSyncClient` and the time we register the callback, and then we would miss a `GsmState` change.

Reply:
Right, it might make races a bit less likely, but the current code also doesn't guarantee race freedom as discussed in another thread.